packages/fx-core/src/plugins/resource/bot/questions.ts

```diff
@@ -2,31 +2,32 @@
 // Licensed under the MIT license.
 import { WayToRegisterBot } from "./enums/wayToRegisterBot";
 import { QuestionNames, QuestionOptions } from "./constants";
-import { NodeType, QTreeNode } from "@microsoft/teamsfx-api";
+import { Inputs, QTreeNode } from "@microsoft/teamsfx-api";
 import isUUID from "validator/lib/isUUID";
 
 const createQuestions = new QTreeNode({
-  type: NodeType.group,
+  type: "group",
 });
 
 const wayToRegisterBotQuestion = new QTreeNode({
   name: QuestionNames.WAY_TO_REGISTER_BOT,
-  type: NodeType.singleSelect,
-  option: QuestionOptions.WAY_TO_REGISTER_BOT_OPTIONS,
+  type: "singleSelect",
+  staticOptions: QuestionOptions.WAY_TO_REGISTER_BOT_OPTIONS,
   title: "Bot registration",
   default: WayToRegisterBot.CreateNew,
   placeholder: "Select an option",
 });
 
 const botIdQuestion = new QTreeNode({
   name: QuestionNames.GET_BOT_ID,
-  type: NodeType.text,
+  type: "text",
   title: "Enter bot id",
   default: "",
   placeholder: "00000000-0000-0000-0000-00000000000",
   prompt: "Open bot managment tool to get bot id",
   validation: {
-    validFunc: async (botId: string) => {
+    validFunc: async (input: string, previousInputs?: Inputs) => {
+      const botId = input as string;
```

Comment:
The as operation is meaningless?

Reply:
yes, previous input is string|string[]|undefined